### PR TITLE
substrate private tx

### LINKED ISSUE #213: [BUG] Substrate VAnchor transaction relaying not catching the TX results
# Overview
Relayer doesn't catch the substrate vanchor tx results, thus the dApp keeps waiting for the relaying tx to be finished 

## Logs
- **[Relayer logs.txt](https://github.com/webb-tools/relayer/files/9045754/logs.txt)**

- **Browser Websocket connection/payload**
 ```
{"substrate":{"vAnchor":{"contract":"","chain":"localnode","id":9,"extData":{"recipient":"5DXNxwFNUKhT1rbYR5ad8hztLutuqCJjkMY4dHXX1Um6qxCJ","relayer":"5GNJqTPyNqANBkUVMN1LPPrxXnFouWXoe2wNSmmEoLctxiZY","extAmount":-1000000000000000,"fee":0,"encryptedOutput1":[239,20,158,247,200,223,164,243,248,43,235,87,89,13,190,173,255,244,198,73,40,163,222,211,106,14,82,74,229,124,132,128,133,53,13,91,197,73,215,114,44,32,45,70,133,137,124,41],"encryptedOutput2":[96,177,166,3,167,29,152,103,151,117,1,241,160,197,90,186,217,10,16,90,34,108,101,146,58,142,146,63,70,122,147,221,161,255,220,47,215,126,165,138,118,195,24,159,103,56,149,55]},"proofData":{"proof":[127,67,222,34,13,148,176,247,180,60,55,222,137,80,82,147,59,242,244,253,51,173,125,91,141,244,38,242,5,245,104,40,90,167,111,93,3,173,90,66,92,108,30,151,151,237,71,172,61,2,121,44,157,22,65,176,179,193,5,34,151,66,49,43,190,71,167,122,74,210,220,179,78,250,47,54,208,71,75,143,139,88,227,206,60,204,106,169,216,140,152,98,33,217,112,41,167,214,25,91,209,14,215,75,138,217,77,72,120,129,199,92,63,211,27,61,40,68,135,45,202,184,40,25,57,238,28,48],"extDataHash":[204,111,86,203,241,17,107,17,70,220,220,25,133,137,57,209,96,238,129,115,237,66,179,10,154,55,86,40,119,78,151,38],"publicAmount":[1,128,57,75,21,104,222,67,145,112,185,121,72,232,51,40,93,88,129,129,182,69,80,184,41,160,49,225,114,78,100,48],"roots":[[120,154,22,37,152,131,145,118,6,60,215,103,39,238,89,89,152,113,141,68,245,211,232,43,43,129,51,92,58,105,95,30],[160,138,218,95,86,180,30,11,21,87,76,148,219,172,9,169,157,121,22,135,145,189,248,188,120,227,71,137,95,88,21,31]],"outputCommitments":[[146,204,29,161,67,172,170,203,90,140,75,252,83,133,95,159,113,25,202,232,126,84,96,71,111,207,57,154,47,63,118,29],[179,202,248,16,150,254,141,227,238,237,63,87,120,129,136,159,96,14,245,217,177,105,44,207,247,4,63,225,212,119,208,19]],"inputNullifiers":[[23,105,73,192,232,207,108,14,185,134,151,31,220,162,242,184,13,174,196,53,14,84,42,202,127,94,213,33,31,14,244,27],[82,90,89,44,137,221,69,204,152,212,102,128,68,79,142,148,0,10,189,201,39,231,46,108,215,79,165,216,126,189,16,26]]}}}}
```
 image

- **Substrate portal**
From the relayer logs the block hash for the tx is `0x7d3e397a12782fc8d4965e5830f43332566ce06e24fb6a098cf93d0567803e59`, and tx hash `0x8eb1e97371d8688ccf941dbf12aa76dc62892ed54dc92acba7d811f9df0eabdd`
image

### PR DESCRIPTION
**Pull request by salman01zp**
## Summary of changes
- [x] Add tracing events to private transaction requests
- [x] Add vanachor private transaction test

### Reference issue to close (if applicable)
- Closes #213 

-----
### Code Checklist 

- [x] Tested
- [x] Documented

### CODE REVIEW

File: tests/lib/webbRelayer.ts [hunk 1]
```diff
@@ -521,6 +510,26 @@ export type EventSelector = {
...
+  relayer: string;
+  extAmount: number;
+  fee: number;
+  encryptedOutput1: number[];
+  encryptedOutput2: number[];
+  refund: number;
+  token: number;
+};
+
+export type SubstrateVanchorProofData = {
```
Comment (drewstone) [inline tests/lib/webbRelayer.ts:524]:
```suggestion
export type SubstrateVAnchorProofData = {
```

File: tests/lib/webbRelayer.ts [hunk 2]
```diff
@@ -521,6 +510,26 @@ export type EventSelector = {
   event?: any;
 };
 
+export type SubstrateVanchorExtData = {
```
Comment (drewstone) [inline tests/lib/webbRelayer.ts:513]:
```suggestion
export type SubstrateVAnchorExtData = {
```

File: tests/test/substrate/vanchorPrivateTransaction.test.ts [hunk 3]
```diff
@@ -0,0 +1,520 @@
...
+  ProvingManagerSetupInput,
+  ArkworksProvingManager,
+  Utxo,
+  calculateTypedChainId,
+  ChainType,
+} from '@webb-tools/sdk-core';
+import { UsageMode } from '@webb-tools/test-utils';
+import {
+  defaultEventsWatcherValue,
+  generateVAnchorNote,
```
Comment (drewstone) [inline tests/test/substrate/vanchorPrivateTransaction.test.ts:52]: Can you make a task to move some of these tools to the test-utils?

**Review by drewstone — CHANGES_REQUESTED** [review]
Looks good! Small nitpicks on the naming

File: tests/test/substrate/vanchorPrivateTransaction.test.ts (hunk 3 above)
Comment (salman01zp) [reply tests/test/substrate/vanchorPrivateTransaction.test.ts:52]: yes, will do that and update the tests soon.

**Pull request merged**